packages/core/src/extension/worker-extension-messenger.ts

```diff
@@ -0,0 +1,29 @@
+import { WorkerEventPoster } from '../common/worker';
+
+import { ExtensionMessage } from './extension-message';
+
+export class WorkerExtensionMessenger {
+    private _workers: { [extensionId: string]: Worker };
+
+    constructor() {
+        this._workers = {};
+    }
+
+    add(extensionId: string, worker: Worker): void {
+        this._workers[extensionId] = worker;
+    }
+
+    post(extensionId: string, message: ExtensionMessage): void {
+        if (!this._workers[extensionId]) {
+            throw new Error(`Worker with extensionId ${extensionId} not found`);
+        }
+
+        const workerPoster = new WorkerEventPoster(this._workers[extensionId], extensionId);
```

Comment:
```suggestion
        const workerPoster = new WorkerEventPoster<ExtensionCommand, ExtensionCommandOrQueryContext>(this._workers[extensionId], extensionId);
```
Maybe we should initialize it this way. Then define new command/query for translation purpose.

Reply:
The rest looks good to me.